File: mobile-client/android/Contract/System/Analytic/AnalyticEventPayload.kt
package app.mobiling.client.contract.system.analytic

/**
 * Canonical contract payload for analytics event emission.
 */
data class AnalyticEventPayload(
    val name: String,
    val attributes: Map<String, String> = emptyMap(),
    val occurredAtIso8601: String? = null,
)

File: client/ios/Contract/System/Analytic/AnalyticEventPayload.swift
import Foundation

public struct AnalyticEventPayload: Sendable, Equatable {
    public let name: String
    public let attributes: [String: String]
    public let occurredAtIso8601: String?

    public init(name: String, attributes: [String: String] = [:], occurredAtIso8601: String? = nil) {
        self.name = name
        self.attributes = attributes
        self.occurredAtIso8601 = occurredAtIso8601
    }
}
